### PR TITLE
[v9] Publish to Release API on release promotion (#15153)

### PR DESCRIPTION
**Pull request by justinas**
Backports https://github.com/gravitational/teleport/pull/15153

Promote pipeline test: https://drone.platform.teleport.sh/gravitational/teleport/14427

Holding off on merging this until the next regular Cloud release (Thursday), as a fix to the server is pending.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -5926,7 +5926,94 @@ volumes:
     claim:
       name: drone-s3-debrepo-pvc
 ---
+kind: pipeline
+type: kubernetes
+name: publish-rlz
+
+environment:
+  RELCLI_BASE_URL: https://releases-staging.platform.teleport.sh
+  RELCLI_IMAGE: 146628656107.dkr.ecr.us-west-2.amazonaws.com/gravitational/relcli:v1.1.65
```
Comment (tigrato) [inline .drone.yml:5935]:
```suggestion
  REGISTRY: 146628656107.dkr.ecr.us-west-2.amazonaws.com
  RELCLI_IMAGE: $REGISTRY/gravitational/relcli:v1.1.65
```

File: .drone.yml [hunk 2]
```diff
@@ -5926,7 +5926,94 @@ volumes:
...
+        from_secret: TELEPORT_BUILD_USER_READ_ONLY_KEY
+      AWS_SECRET_ACCESS_KEY:
+        from_secret: TELEPORT_BUILD_USER_READ_ONLY_SECRET
+      AWS_DEFAULT_REGION: us-west-2
+    volumes:
+      - name: dockersock
+        path: /var/run
+    commands:
+      - apk add --no-cache aws-cli
+      - aws ecr get-login-password | docker login -u="AWS" --password-stdin 146628656107.dkr.ecr.us-west-2.amazonaws.com
```
Comment (tigrato) [inline .drone.yml:5971]:
```suggestion
      - aws ecr get-login-password | docker login -u="AWS" --password-stdin $REGISTRY
```

Comment (justinas) [reply .drone.yml:5971]: These are good suggestions, but perhaps let's keep them for the next batch of changes, as these are backports and I would like to keep them consistent with what's been merged to master.

Comment (tigrato) [reply .drone.yml:5971]: Makes sense

**Review by tigrato — APPROVED** [review]

**Review by strideynet — APPROVED** [review]

**Pull request merged**